### ISSUE TITLE
All labeled examples are as part of unlabeled data without using their labels?

### ISSUE THREAD

**Issue by Chen-Song** [kekmodel/FixMatch-pytorch#10]
Hi, in the paper, the authors say that they include all labeled examples as part of unlabeled data
without using their labels when constructing U. However, in your code, in 106 line of cifar.py, it is unlabeled_idx.extend(idx[label_per_class:]). Is this a mistake？Thanks

**Comment by jnyborg** [comment]
I was also wondering about this and found that the original code do include labelled examples as part of the unlabeled data, so I believe it is a mistake. See [create_unlabeled.py](https://github.com/google-research/fixmatch/blob/master/scripts/create_unlabeled.py) and [create_split.py](https://github.com/google-research/fixmatch/blob/master/scripts/create_split.py)

**Comment by Chen-Song** [comment]
> I was also wondering about this and found that the original code do include labelled examples as part of the unlabeled data, so I believe it is a mistake. See [create_unlabeled.py](https://github.com/google-research/fixmatch/blob/master/scripts/create_unlabeled.py) and [create_split.py](https://github.com/google-research/fixmatch/blob/master/scripts/create_split.py)

Hi, do you know why the code copies the labeled data to 65536, and the unlabeled data is still 7 times larger (65536*7). I don't quite understand why this is necessary.

**Comment by jnyborg** [comment]
If you refer to the logic [here](https://github.com/kekmodel/FixMatch-pytorch/blob/9044f2ef656fb135cf93c9b9f0fe1f6870ea266f/dataset/cifar.py#L110), no, I'm not sure why that is necessary either. Maybe the original author can pitch in.

**Comment by Chen-Song** [comment]
> If you refer to the logic [here](https://github.com/kekmodel/FixMatch-pytorch/blob/9044f2ef656fb135cf93c9b9f0fe1f6870ea266f/dataset/cifar.py#L110), no, I'm not sure why that is necessary either. Maybe the original author can pitch in.

Hi, I recently read FixMatch's paper and code? In the implementation of FixMatch, for example, 4000 labeled data; you need to copy the labeled data to 65536, and copy the remaining unlabeled to 65536*7. I don't quite understand why do this. Do you know why do this？ Or is it the default setting for semi-supervised fields?

**Comment by YBZh** [comment]
> > If you refer to the logic [here](https://github.com/kekmodel/FixMatch-pytorch/blob/9044f2ef656fb135cf93c9b9f0fe1f6870ea266f/dataset/cifar.py#L110), no, I'm not sure why that is necessary either. Maybe the original author can pitch in.
> 
> Hi, I recently read FixMatch's paper and code? In the implementation of FixMatch, for example, 4000 labeled data; you need to copy the labeled data to 65536, and copy the remaining unlabeled to 65536*7. I don't quite understand why do this. Do you know why do this？ Or is it the default setting for semi-supervised fields?

It seems that they prepare the training data of a whole epoch (batch size 64 * iteration 1024 = 65536) at the beginning of the epoch.

**Comment by kekmodel** [comment]
Thanks for letting me know. I'll fix it!
